app.py

```diff
@@ -163,6 +209,94 @@ def list_data_objects(**kwargs):
         'next_page_token': next_page_token}
 
 
+@app.route("{}/dataobjects/{}".format(base_path, "{data_object_id}"),
+           methods=['PUT'], cors=True)
+def update_data_object(data_object_id):
+    """
+    Updates a Data Object's alias field only, while not modifying
+    version information.
+
+    :param kwargs:
+    :return:
+    """
+    # Before anything, make sure they are allowed to make the
+    # modification.
+    if not check_auth():
+        return Response({'msg': 'Not authorized to access '
+                                'this service. Set the '
+                                'access_token in request '
+                                'headers.'}, status_code=403)
+
+    # First try to get the Object specified
+    data_object, source = safe_get_data_object(data_object_id)
+
+    # Now check to see the contents don't already contain
+    # any aliases we want to add.
+
+    update_body = app.current_request.json_body
+
+    new_aliases = filter(
+        lambda x: x not in data_object['aliases'],
+        update_body['aliases'])
+    # if len(new_aliases) == 0:
+    #     return Response(
+    #         {'msg': 'No new aliases, nothing was changed. '
+    #                 'Please check your request details.'}, status_code=400)
+
+    data_object['aliases'] = data_object['aliases'] + new_aliases
+
+    es_id = source['_id']
+
+    # It is an implementation detail of this DOS that aliases
+    # are namespaced to provide a categorical discovery
+    # process.
+
+    # We are expecting string keys as aliases, so we can add
+    # them if they are not already present.
+
+    for alias in new_aliases:
+        if ':' not in alias:
+            return Response(
+                {'msg': 'Aliases must be namespaced by providing'
+                        'a {key}:{value} structure'}, status_code=400)
+
+    new_tuples = map(lambda x: x.split(':'), new_aliases)
+
+    # But first, to avoid overwriting existing keys, we check
+    # against the contents of the source document.
+
+    existing_keys = source['_source'].keys()
+
+    for new_tuple in new_tuples:
+        if new_tuple[0] in existing_keys and new_tuple[0] not in safe_keys:
+            return Response(
+                {'msg': 'Existing keys can\'t be overwritten.'
+                        'Please check the alias {} is not already '
+                        'in the source document.'.format(new_tuple)},
+                status_code=400)
+
+    # Now that we believe we are not overwriting an existing
+    # key and that we have the source document, we can
+    # attempt to perform the update.
+
+    updated_fields = {x[0]: x[1] for x in new_tuples}
+
+    es_update_response = client.make_request(
```

Comment:
Ideally this should check the ES document version to identify/handle concurrent updates.
https://www.elastic.co/guide/en/elasticsearch/guide/current/optimistic-concurrency-control.html

There are examples of the use of this in the HCA data-store primary index code.